src/bun.js/bindings/exports.zig

```diff
@@ -2130,6 +2130,12 @@ pub const ZigConsoleClient = struct {
                     }
                 },
                 .Integer => {
+                    if (value.isSymbol() or value.isUndefined()) {
+                        this.addForNewLine("NaN".len);
+                        writer.print(comptime Output.prettyFmt("<r><yellow>NaN<r>", enable_ansi_colors), .{});
+                        return;
+                    }
+
                     const int = value.coerce(i64, this.globalThis);
```

Comment:
I got a problem here and need some help, coerce to `i64` won't  return `NaN`.
So if we try: `console.log("%d", "one")`, it will print `0`.

Two problem here:

1. coerceToInt64

When we pass a string like `one`, it will finally call `value.toInt32`.

https://github.com/oven-sh/bun/blob/c2cd7402a03f8dfd10c9f77facf108e626e2ac83/src/bun.js/bindings/bindings.cpp#L3539-L3556

2. [value.toInt32](https://github.com/oven-sh/WebKit/blob/7cd84abfa787e4b96b27c3ef1e28a4eb1aa49aa1/Source/JavaScriptCore/runtime/JSCJSValueInlines.h#L45-L52)

Then, `toNumber` will return a `NaN`, but it will be covert to 0 in the `JSC::toInt32`.

```cpp
ALWAYS_INLINE int32_t JSValue::toInt32(JSGlobalObject* globalObject) const
{
    if (isInt32())
        return asInt32();

    double d = toNumber(globalObject);
    return JSC::toInt32(d);
}
```

I am not familiar of cpp, how can I return a NaN in a int type? Or can we return a maxInt to represent the NaN?
As far as I know, NaN will be and only will a float.

@paperdave

Reply:
Or I have a simple solution for it:

```suggestion
                    const int = value.coerce(i64, this.globalThis);
                    if (value.isString() and int == 0) {
                        var str = ZigString.Empty;
                        value.toZigString(&str, this.globalThis);

                        if (!str.eql(ZigString.init("0"))) {
                            this.addForNewLine("NaN".len);
                            writer.print(comptime Output.prettyFmt("<r><yellow>NaN<r>", enable_ansi_colors), .{});
                            return;
                        }
                    }
```

Reply:
we should follow the spec, which unfortunately says to call `parseInt`. here is what i am thinking after messing with this for a little bit

```zig
// https://console.spec.whatwg.org/#formatter

// 1. If Type(current) is Symbol, let converted be NaN
// 2. Otherwise, let converted be the result of Call(%parseInt%, undefined, current, 10)
const int: i64 = brk: {
    if (value.isInt32()) {
        // Already an int
        break :brk value.asInt32();
    }
    double_convert: if (value.isNumber() or !value.isSymbol()) {
        var value = value.coerceToDouble();

        if (!std.math.isFinite(value)) {
            // for NaN and the string Infinity and -Infinity, parseInt returns NaN
            break :double_convert;
        }

        // simulate parseInt, which converts the argument to a string and
        // then back to a number, without converting it to a string
        const max_before_e_notation = 1000000000000000000000;
        const min_before_e_notation = 0.000001;
        if (value > 0) {
            if (value >= max_before_e_notation) {
                // toString prints 0.000+e0, which parseInt will
                var int: i64 = @floor(value);
                while (value > 10) value = @divFloor(int, 10);
                break :brk 0;
            } else if (value <= min_before_e_notation) {
                // todo
            }
        } else if (value <= -max_before_e_notation) {
            // todo
        } else if (value >= min_before_e_notation) {
            // todo
        }

        // 0000.00 float, floor would be the same as parseInt
        break :brk @floor(value);
    }
    // for NaN and the string Infinity and -Infinity, parseInt returns NaN
    this.addForNewLine("NaN".len);
    writer.print(comptime Output.prettyFmt("<r><yellow>NaN<r>", enable_ansi_colors), .{});
};
```
not fully done but the goal is to keep this as fast as possible

i wonder if we should split .Integer into two types, .ConvertToInteger for the `%i` specifier, and then .Integer for the "yeah, this is always an integer"

Reply:
this might need a separate case for strings where we just call `parseInt` on the string